### ISSUE TITLE
[WinError 127] The specified procedure could not be found. 

### ISSUE THREAD

**Issue by talhaanwarch** [awslabs/autogluon#1260]
If i call `from autogluon.vision import ImagePredictor` in first place, I got this error.
`OSError: [WinError 127] The specified procedure could not be found. Error loading "d:\env\autog\lib\site-packages\torch\lib\caffe2_detectron_ops_gpu.dll" or one of its dependencies.`

```
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_11800/3281375176.py in <module>
----> 1 from autogluon.vision import ImagePredictor

d:\env\autog\lib\site-packages\autogluon\vision\__init__.py in <module>
----> 1 from .predictor import ImagePredictor
      2 from .detector import ObjectDetector
      3 
      4 ImageDataset = ImagePredictor.Dataset
      5 ImageDetectionDataset = ObjectDetector.Dataset

d:\env\autog\lib\site-packages\autogluon\vision\predictor\__init__.py in <module>
----> 1 from .predictor import *

d:\env\autog\lib\site-packages\autogluon\vision\predictor\predictor.py in <module>
      8 import numpy as np
      9 import pandas as pd
---> 10 from gluoncv.auto.tasks import ImagePrediction as _ImageClassification
     11 from gluoncv.model_zoo import get_model_list
     12 

d:\env\autog\lib\site-packages\gluoncv\__init__.py in <module>
     25 
     26 try:
---> 27     _require_pytorch_version('1.4.0', '2.0.0')
     28     _found_pytorch = True
     29 except ImportError:

d:\env\autog\lib\site-packages\gluoncv\check.py in _require_pytorch_version(torch_version, max_torch_version)
     26 def _require_pytorch_version(torch_version, max_torch_version='2.0.0'):
     27     try:
---> 28         import torch
     29         from distutils.version import LooseVersion
     30         if LooseVersion(torch.__version__) < LooseVersion(torch_version) or \

d:\env\autog\lib\site-packages\torch\__init__.py in <module>
    115                 err = ctypes.WinError(last_error)
    116                 err.strerror += f' Error loading "{dll}" or one of its dependencies.'
--> 117                 raise err
    118             elif res is not None:
    119                 is_loaded = True

```

after this if I call `import torch` i got the following error
```
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_11800/4265195184.py in <module>
----> 1 import torch

d:\env\autog\lib\site-packages\torch\__init__.py in <module>
    115                 err = ctypes.WinError(last_error)
    116                 err.strerror += f' Error loading "{dll}" or one of its dependencies.'
--> 117                 raise err
    118             elif res is not None:
    119                 is_loaded = True

OSError: [WinError 127] The specified procedure could not be found. Error loading "d:\env\autog\lib\site-packages\torch\lib\caffe2_detectron_ops_gpu.dll" or one of its dependencies.
```

But if I import torch first then ` autogluon.vision`, there is no error
```
import torch
from autogluon.vision import ImagePredictor

```
works fine, but
```
from autogluon.vision import ImagePredictor
import torch
```
does not work

```
OS: Windows 11
GPU: GTX 1060
CUDA: 10.1
AG version: '0.2.1b20210729'
```

**Comment by Innixma** [comment]
@zhreshold looks like a bug in GluonCV, could you take a look?

**Comment by zhreshold** [comment]
@talhaanwarch this is the error in pytorch, checkout the similar issues here: https://github.com/pytorch/pytorch/issues/37733

Regarding the import order, it's possible that there are more than one torch in you path so altering the order have different behavior when dynamic libs were loaded.

**Comment by yinweisu** [comment]
Close this issue as this should been solved after 0.3 release.
